nimble/host/src/ble_hs_hci.c

```diff
@@ -339,11 +339,13 @@ ble_hs_hci_cmd_tx(uint16_t opcode, const void *cmd, uint8_t cmd_len,
 static void
 ble_hs_hci_rx_ack(uint8_t *ack_ev)
 {
+#if 0
     if (ble_npl_sem_get_count(&ble_hs_hci_sem) > 0) {
         /* This ack is unexpected; ignore it. */
         ble_hci_trans_buf_free(ack_ev);
         return;
     }
+#endif
```

Comment:
This is ifdefed since it is a race condition

Reply:
could you comment on this a bit? it looks like this code was here since day 1

Reply:
I found that this check would return true unexpectedly and it would not get to the release. It seemed to be a race condition due to the fact that the this function would get called too fast under certain conditions. I think the sequence was that the lock was actually set after the if check.

Reply:
Ok, I looked further into the issue and it was not a race condition after all. The problem is that ble_npl_sem_get_count() returns unsigned integer and NuttX returns negative numbers in sem_getvalue() to indicate number of lock count (this is a valid option as per the man page). So this would generate 65536 as the sem value and the if would get triggered.
It now works as expected.